File: rxpresenter/src/main/java/com/ezhome/rxpresenter/reactive/DefaultSubscriber.java
package com.ezhome.rxpresenter.reactive;

import rx.Subscriber;
import timber.log.Timber;

/**
 * Default subscriber base class to be used whenever you want default error handling.
 * This used for RxJava2 {@link rx.Observable}
 */
public class DefaultSubscriber<T> extends Subscriber<T> {

  @Override public void onNext(T t) {
    // no-op by default.
  }

  @Override public void onCompleted() {
    //no-op by default
  }

  @Override public void onError(Throwable e) {
    Timber.e(e, e.getMessage());
  }
}

File: rxpresenter-kotlin/src/main/kotlin/com/ezhome/rxpresenter/kotlin/reactive/DefaultSubscriber.kt
package com.ezhome.rxpresenter.reactive

import rx.Subscriber
import timber.log.Timber

/**
 * Default subscriber base class to be used whenever you want default error handling.
 * This used for RxJava2 [rx.Observable]
 */
class DefaultSubscriber<T> : Subscriber<T>() {

    @Override
    override fun onNext(t: T) {
        // no-op by default.
    }

    @Override
    override fun onCompleted() {
        //no-op by default
    }

    @Override
    override fun onError(e: Throwable) {
        Timber.e(e, e.message)
    }
}